exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/DropTableHandler.java

```diff
@@ -69,6 +71,15 @@ public PhysicalPlan getPlan(SqlNode sqlNode) throws ValidationException, RelConv
           .build(logger);
     }
 
+    final Table existingTable = SqlHandlerUtil.getTableFromSchema(drillSchema, tableName);
```

Comment:
Can you confirm that `getTableFromSchema(...)` returns `null` only if the table does not exist?